core/delete_area.js

```diff
@@ -49,5 +58,58 @@ Blockly.DeleteArea.prototype.wouldDeleteBlock = function(_block, couldConnect) {
  *     this area.
  */
 Blockly.DeleteArea.prototype.wouldDeleteBubble = function(_bubble) {
+  this.wouldDelete_ = true;
   return true;
 };
+
+/**
+ * Returns whether the provided block should not be moved after being dropped
+ * on this component. If true, block will return to where it was when the drag
+ * started.
+ * @param {!Blockly.BlockSvg} _block The block.
+ * @return {boolean} Whether the block provided should be returned to drag
+ *     start.
+ */
+Blockly.DeleteArea.prototype.shouldPreventBlockMove = function(_block) {
+  return false;
+};
+
+/**
+ * Returns whether the provided bubble should not be moved after being dropped
+ * on this component. If true, bubble will return to where it was when the drag
+ * started.
+ * @param {!Blockly.IBubble} _bubble The bubble.
+ * @return {boolean} Whether the bubble provided should be returned to drag
+ *    start.
+ */
+Blockly.DeleteArea.prototype.shouldPreventBubbleMove = function(_bubble) {
+  return false;
+};
+
+/**
+ * Handles when a cursor with a block or bubble exits this drag target.
+ * @override
+ */
+Blockly.DeleteArea.prototype.onDragExit = function() {
+  this.wouldDelete_ = false;
+};
+
+/**
+ * Handles when a block is dropped on this component. Should not handle delete
+ * here.
+ * @param {!Blockly.BlockSvg} _block The block.
+ * @override
+ */
+Blockly.DeleteArea.prototype.onBlockDrop = function(_block) {
+  this.wouldDelete_ = false;
```

Comment:
They could do a super call last (super call only needs to be first in the constructor), but I agree it is a bit confusing. I'll think a bit about this.

Reply:
I've pushed a new version that moves the responsibility of whether a block/bubble is deletable to the `wouldDeleteBlock`/`wouldDeleteBubble` implementation. This allows for me to only need to set `this.wouldDelete_` in those methods because I can guarantee that these methods called before drag enter/exit/over.